### PR TITLE
HTS-723: Added functionality for stride eligibility-check and get-paye-personal-details

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

**Comment by sureshhmrc** [comment]
please go through `https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?spaceKey=Repayments&title=Solution+Designs+and+API+specs&preview=/71125105/76489739/DES%20API_%2BGet%2BPAYE%2BPerson%2BDetails%2B-%2Bv4.4%20-%2020160921_1029.docx`

File: app/uk/gov/hmrc/helptosave/controllers/StrideController.scala [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+              Ok(Json.toJson(EligibilityResponseHolder(r)))
+          }
+        )
+      }
+    )
+  }
+
+  def getPayePersonalDetails(ninoParam: String): Action[AnyContent] = authorisedFromStride { implicit request ⇒
+
+    Try(new String(base64Decoder.decode(ninoParam))).fold(
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/controllers/StrideController.scala]: there's some common code that can be factored out here, perhaps something like this:
```scala
def withBase64DecodedNINO(f: NINO => Future[Result])(implicit request: Request[AnyContent]): Future[Result] =
  Try(new String(base64Decoder.decode(ninoParam))).fold(
    { error ⇒
        logger.warn(s"Could not decode nino from encrypted param: $error")
        InternalServerError
      }, f
  )
```

File: test/uk/gov/hmrc/helptosave/connectors/PayePersonalDetailsConnectorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+class PayePersonalDetailsConnectorSpec
+  extends TestSupport
+  with GeneratorDrivenPropertyChecks
+  with ServicesConfig
+  with MockPagerDuty
+  with TestData {
+
+  MdcLoggingExecutionContext
```
Comment (AndyHWChung) [inline test/uk/gov/hmrc/helptosave/connectors/PayePersonalDetailsConnectorSpec.scala]: do we need this floating around?

File: app/uk/gov/hmrc/helptosave/models/PayePersonalDetails.scala [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import play.api.libs.json.Reads.localDateReads
+import play.api.libs.json.Writes.temporalWrites
+import play.api.libs.json._
+
+case class PayePersonalDetails(name:        Name,
+                               dateOfBirth: LocalDate,
+                               address:     Address,
+                               phoneNumber: Option[TelePhoneNumber])
+
+case class Name(title:                   Option[String],
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/models/PayePersonalDetails.scala]: don't think we need the title

File: app/uk/gov/hmrc/helptosave/models/PayePersonalDetails.scala [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import java.time.format.DateTimeFormatter
+
+import play.api.libs.json.Reads.localDateReads
+import play.api.libs.json.Writes.temporalWrites
+import play.api.libs.json._
+
+case class PayePersonalDetails(name:        Name,
+                               dateOfBirth: LocalDate,
+                               address:     Address,
+                               phoneNumber: Option[TelePhoneNumber])
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/models/PayePersonalDetails.scala]: we can proabbly ignore phone numbers for now

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**